debug/testlib.py

```diff
@@ -1440,6 +1440,29 @@ def parkOtherHarts(self, symbol=None):
         self.gdb.select_hart(self.hart)
         self.gdb.command(f"monitor targets {self.hart.id}")
 
+    def set_pmp_deny(self, address, size=4 * 1024):
+        # Enable physical memory protection, no permission to access specific
+        # address range (default 4KB).
+        try:
+            self.gdb.p("$mseccfg=0x4")  # RLB
+            self.gdb.p("$pmpcfg0=0x98") # L, NAPOT, !R, !W, !X
+            self.gdb.p("$pmpaddr0="
+                           f"0x{((address >> 2) | ((size - 1) >> 3)):x}")
+            self.gdb.command("monitor riscv exec_progbuf 0x12000073") # 0x12000073 is sfence.vma
+        except CouldNotFetch:
+            # PMP registers are optional
+            pass
```

Comment:
1. can you add some indication in the log that an attempt resulted in a some kind of exception?
2. do you really need this `try` block since you've introduced this ` support_set_pmp_deny `  property for the target?